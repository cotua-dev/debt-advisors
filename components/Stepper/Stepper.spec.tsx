import { render, RenderResult } from '@testing-library/react';
import { Stepper } from './index';

describe('Stepper Component', (): void => {
    let renderResult: RenderResult;

    beforeAll((): RenderResult => renderResult = render(<Stepper/>));

    test('renders', (): void => {
        const { container } = renderResult;
        expect(container.firstChild).toMatchSnapshot();
    });

    afterAll((): void => renderResult.unmount());
});
