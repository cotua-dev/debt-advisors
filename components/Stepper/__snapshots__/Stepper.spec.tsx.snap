// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper Component renders 1`] = `
<section
  class="debt-advisors-stepper"
>
  <div>
    <small>
      1 of 11
    </small>
  </div>
  <div
    class="steps-wrapper"
  >
    <div
      class="step "
    >
      <strong
        class="question"
      >
        What are you looking to do?
      </strong>
      <div
        class="multiple-choice-wrapper"
      >
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Reduce your monthly payments
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Consolidate your payments
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Reduce your interest rates
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            All of the above
          </strong>
        </button>
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        What kind of debts do you have?
      </strong>
      <div
        class="multiple-choice-wrapper"
      >
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Credit cards
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Personal loans
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Medical bills
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            All of the above
          </strong>
        </button>
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        Are you behind on your payments?
      </strong>
      <div
        class="multiple-choice-wrapper"
      >
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Yes - more than 60 days
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Yes - more than 30 days
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            No
          </strong>
        </button>
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        What made you fall behind?
      </strong>
      <div
        class="multiple-choice-wrapper"
      >
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Loss of income
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Increase in cost of living
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Unexpected expenses
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Medical hardship
          </strong>
        </button>
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        What is your monthly income?
      </strong>
      <div
        class="currency-field-wrapper"
      >
        <input
          class="currency-field"
          inputmode="numeric"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        What is the total amount of your unsecured debt?
      </strong>
      <div
        class="currency-field-wrapper"
      >
        <input
          class="currency-field"
          inputmode="numeric"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        What city are you in?
      </strong>
      <div
        class="location-wrapper"
      >
        <input
          class="location-field"
          value=""
        />
        <picture>
          <source />
          <img
            alt="Powered By Google Logo"
            class="powered-by-google-logo"
            loading="lazy"
          />
        </picture>
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        What is your name?
      </strong>
      <div
        class="name-fields-wrapper"
      >
        <div
          class="name-field-wrapper"
        >
          <label
            class="name-field-label"
            for="firstName"
          >
            First Name
          </label>
          <input
            class="name-field"
            id="firstName"
            name="firstName"
            type="text"
            value=""
          />
        </div>
        <div
          class="name-field-wrapper"
        >
          <label
            class="name-field-label"
            for="lastName"
          >
            Last Name
          </label>
          <input
            class="name-field"
            id="lastName"
            name="lastName"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        What is your email?
      </strong>
      <div
        class="email-field-wrapper"
      >
        <input
          class="email-field"
          id="email"
          name="email"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        Please provide your phone number for verification
      </strong>
      <div
        class="phone-field-wrapper"
      >
        <input
          class="phone-field"
          id="phone"
          inputmode="numeric"
          name="phone"
          type="tel"
          value=""
        />
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        Please provide the code sent to your phone number
      </strong>
      <div
        class="code-field-wrapper"
      >
        <input
          class="code-field"
          disabled=""
          id="code"
          inputmode="numeric"
          name="code"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    class="stepper-controls-wrapper"
  >
    <button
      class="previous-button"
      disabled=""
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-angle-left fa-w-8 caret"
        data-icon="angle-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      class="next-button"
      disabled=""
      type="button"
    >
      Next
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-angle-right fa-w-8 caret"
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</section>
`;
